app-backend/batch/src/main/scala/com/azavea/rf/batch/sentinel2/ImportSentinel2.scala

```diff
@@ -0,0 +1,284 @@
+package com.azavea.rf.batch.sentinel2
+
+import com.azavea.rf.batch.Job
+import com.azavea.rf.batch.util._
+import com.azavea.rf.database.{Database => DB}
+import com.azavea.rf.database.tables.{Scenes, Users}
+import com.azavea.rf.datamodel._
+import com.azavea.rf.batch.util._
+
+import io.circe._
+import io.circe.syntax._
+import cats.implicits._
+import geotrellis.proj4.CRS
+import geotrellis.slick.Projected
+import geotrellis.vector._
+import org.postgresql.util.PSQLException
+
+import java.security.InvalidParameterException
+import java.time.{LocalDate, ZoneOffset, ZonedDateTime}
+import java.util.UUID
+
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+case class ImportSentinel2(startDate: LocalDate = LocalDate.now(ZoneOffset.UTC))(implicit val database: DB) extends Job {
+  val name = ImportSentinel2.name
+
+  /** Required for sceneMetadata collection, to unbox Options */
+  implicit def optionToString(opt: Option[String]): String = opt.getOrElse("")
+
+  /** Get S3 client per each call */
+  def s3Client = S3(sentinel2Config.awsRegion)
+
+  def createImages(sceneId: UUID, tileInfo: Json, resolution: Float): List[Image.Banded] = {
+    val tileInfoPath = tileInfo.hcursor.downField("path").as[String].toOption.getOrElse("")
+    logger.info(s"Creating images for $tileInfoPath with resolution $resolution")
+
+    val keys = resolution match {
+      case 10f =>
+        List(
+          s3Client.getObject(sentinel2Config.bucketName, s"${tileInfoPath}/B02.jp2"),
```

Comment:
Ah ok, so I see we need the metadata for the length. I think it would be better to do just that then. The following would work:

```scala
val keys = resolution match {
      case 10f =>
        List(
          s"${tileInfoPath}/B02.jp2",
          s"${tileInfoPath}/B03.jp2",
          s"${tileInfoPath}/B04.jp2",
          s"${tileInfoPath}/B08.jp2"
        )
      case 20f =>
        List(
          s"${tileInfoPath}/B05.jp2",
          s"${tileInfoPath}/B06.jp2",
          s"${tileInfoPath}/B07.jp2",
          s"${tileInfoPath}/B8A.jp2"
        )
      case 60f =>
        List(
          s"${tileInfoPath}/B01.jp2",
          s"${tileInfoPath}/B09.jp2",
          s"${tileInfoPath}/B10.jp2"
        )
      case _ =>
        throw new InvalidParameterException(s"Unable to create images for $tileInfoPath at resolution $resolution")
    }

    keys.map { obj =>
      val filename = obj.split("/").last
      val imageBand = filename.split("\\.").head
      val objMetadata = s3Client.client.getObjectMetadata(sentinel2Config.bucketName, obj)
      Image.Banded(
        organizationId   = sentinel2Config.organizationUUID,
        rawDataBytes     = objMetadata.getContentLength.toInt,
        visibility       = Visibility.Public,
        filename         = filename,
        sourceUri        = s"${sentinel2Config.baseHttpPath}/${obj}",
        owner            = airflowUser.some,
        scene            = sceneId,
        imageMetadata    = Json.Null,
        resolutionMeters = resolution,
        metadataFiles    = List(),
        bands            = Seq(sentinel2Config.bandByName(imageBand)).flatten
      )
    }
```